### ISSUE TITLE
Mutators not working as expected

### ISSUE THREAD

**Issue by kaidjohnson** [final-form/final-form#240]
### Are you submitting a **bug report** or a **feature request**?

Bug Report

### What is the current behavior?

Triggering a mutator on a field before it is registered results in the mutation not being set. Triggering a mutator after the field has been registered results the first subscriber callback to not have the mutation, but subsequent callbacks to have it.

This behavior is expected given how mutators work, however the documentation is misleading.

Given the example from https://github.com/final-form/final-form-set-field-data

```
form.mutators.setFieldData('firstName', { awesome: true })

form.registerField(
  'firstName',
  fieldState => {
    const { awesome } = fieldState.data // true
  },
  {
    // ...other subscription items
    data: true
  }
)
```

We're expecting `awesome` to be `true`, however, it is `undefined`, as seen here:

https://codepen.io/ThunderheadONE/pen/NZEBGe?editors=1112

If we move the mutator to after the registerField, we get one churn with `undefined` (before the mutator), followed by a second update (triggered by the mutator itself) with the expected value of `true`.

This issue might be related to [#178]

### What is the expected behavior?

At the very least, the documentation probably needs to be updated with a working example and potentially get a few additional complex examples added. 

### Sandbox Link

https://codepen.io/ThunderheadONE/pen/NZEBGe?editors=1112

**Comment by Syynth** [comment]
I think this is probably just a documentation issue around `final-form-set-field-data` as the entire implementation of that mutator is just
```javascript
const setFieldData: Mutator = (args: any[], state: MutableState) => {
  const [name, data] = args
  const field = state.fields[name]
  if (field) {
    field.data = { ...field.data, ...data }
  }
}
```
so it looks pretty intentional that it's dropping the data because the field has not yet been registered. I suppose you could create your own field information on the state, but the logic is quite complex (relatively speaking) in that it is over 100 lines and manages field subscribers and validation setup that the form needs.

A very simple change that could be made would be to extend the `formApi.registerField`'s 4th argument (`fieldConfig`) to also accept a `data` attribute. This would actually be a one line change at [L764](https://github.com/final-form/final-form/blob/master/src/FinalForm.js#L764), changing `data: {}` to `data: fieldConfig && fieldConfig.data || {}`.

**Comment by erikras** [comment]
Published @Syynth's fix in [`v4.16.2`](https://github.com/final-form/final-form/releases/tag/v4.16.2). You can now provide an initial `data` value when you register your field.
